core/trino-main/src/main/java/io/trino/sql/planner/OptimizerConfig.java

```diff
@@ -471,15 +473,15 @@ public OptimizerConfig setOptimizeMetadataQueries(boolean optimizeMetadataQuerie
         return this;
     }
 
-    public boolean isUseMarkDistinct()
+    public UseMarkDistinct getUseMarkDistinct()
     {
         return useMarkDistinct;
     }
 
     @Config("optimizer.use-mark-distinct")
-    public OptimizerConfig setUseMarkDistinct(boolean value)
+    public OptimizerConfig setUseMarkDistinct(String value)
```

Comment:
This is backward incompatible change.

We should add another property:

```
optimizer.mark-distinct-strategy=NONE|ALWAYS|AUTOMATIC
```
(consistent with `optimizer.join-reordering-strategy`). 

Then we can map `optimizer.use-mark-distinct=true` -> `optimizer.mark-distinct-strategy=AUTOMATIC`
and `optimizer.use-mark-distinct=true` -> `optimizer.mark-distinct-strategy=NONE`.
We can also mark `optimizer.use-mark-distinct` as deprecated.

Reply:
This brings some problems because now we have two properties controlling single behavior. What if someone sets `use_mark_distinct= false` and `optimizer.mark-distinct-strategy=ALWAYS`?
Should we force users to match? what if user wants optimizer.mark-distinct-strategy=ALWAYS?

Reply:
You can make the old one `nullabe` to check if it was set explicitly and prevent invalid config

Reply:
added mark-distinct-strategy